src/basic/log.c

```diff
@@ -353,7 +353,7 @@ void log_forget_fds(void) {
 }
 
 void log_set_max_level_realm(LogRealm realm, int level) {
-        assert((level & LOG_PRIMASK) == level);
+        assert(FLAGS_SET(LOG_PRIMASK, level));
```

Comment:
hmm, this is a weird one. This check was added here to validate that the specifier level argument has no bits set but the level. With the change now it reads a bit weirdly, as in "check that `LOG_PRIMASK` has all bits in `level` set", which is still correct, but a weird way to express what we want to check here...

maybe the fix here is to change the expressio to:

```c
assert(l(evel & ~LOG_PRIMASK) == 0);
```
which is equivalent, but I think more explicitly as it denotes exactly: all other bits than the ones from LOG_PRIMASK should be zero.